docusaurus/docs/dev-docs/upgrade-tool.md

```diff
@@ -0,0 +1,164 @@
+
+The upgrade tool is a Strapi package and can be run from the CLI.
+
+## Version types
+
+Strapi version numbers respect the [semantic versioning](https://semver.org/) conventions:
+
+<ThemedImage
+  alt="Version numbers explained"
+  sources={{
+    light: '/img/assets/update-migration/version-numbers.png',
+    dark: '/img/assets/update-migration/version-numbers_DARK.png',
+  }}
+/>
+
+- The first number is the **major** version number.
+- The second number is the **minor** version number.
+- The third number is the **patch** version number.
+
+The upgrade tool allows upgrading to a major, minor, or patch version.
+
+| The latest Strapi v4 version is v4.16.2 and my Strapi application is currently on… | If I run… | My Strapi application will be upgraded to … |
+|----|----|----|
+| v4.14.1 | `npx @strapi/upgrade patch` | v4.14.6<br/><br/>(because v4.14.6 is the latest patch version for the v4.14 minor version) |
+| v4.14.1 | `npx @strapi/upgrade minor` | v4.16.2 |
+| v4.14.1 | `npx @strapi/upgrade major` | Nothing.<br/><br/>I first need to run `npx @strapi/upgrade minor` to upgrade to v4.16.2. |
+| v4.16.2 | `npx @strapi/upgrade major` | v5.0.0  |
+
+## Upgrade to a major version
+
+Run the upgrade tool with the `major` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade major
+```
+
+During the upgrade process, the application dependencies are updated and installed, and the codemods up until the targeted major version are executed.
+
+:::note
+If your application is not already running the latest minor and patch version in the current major, the `major` upgrade is prevented, and you will first need to upgrade to the latest [minor.patch](#upgrade-to-a-minor-version) version in the current major version. This means that moving from v4.14.4 to v5.0.0 is a 2-step process because the latest v4 version is v4.16.2.
+:::
+
+## Upgrade to a minor version
+
+Run the upgrade tool with the `minor` parameter to upgrade the project to the latest minor and patch version of Strapi:
+
+```bash
+npx @strapi/upgrade minor
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Upgrade to a patch version
+
+Run the upgrade tool with the `patch` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade patch
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Run codemods only
+
+Run the upgrade tool with the `codemods` parameter to execute a utility that allows selecting the codemods to be executed. With this command, only the codemods are run, the dependencies are not updated nor installed.
+
+```bash
+npx @strapi/upgrade codemods
+```
+
+## Options
+
+The `npx @strapi/upgrade` command can accept the following options:
```

Comment:
To nitpick a bit, the upgrade commands* can accept options, the upgrade CLI itself only has commands (major/minor/patch)

Reply:
Thank you, makes sense! I wasn't sure how to write but had a feeling it was not 100% correct :D